### ISSUE TITLE
NMP data check improvements

### ISSUE THREAD

**Issue by michihdeu** [TravelMapping/Web#187]
Near miss points (NMP) are waypoints which are near to each other and should be checked by region maintainers if they can be merged. The [.nmp file](http://tm.teresco.org/graphs/tm-master.nmp) containing NMP pairs and [log file](http://tm.teresco.org/logs/nearmisspoints.log) are automatically created with each site update. False positives (FPs) are entered in a [different log file](https://github.com/TravelMapping/HighwayData/blob/master/nmpfps.log) by maintainers.

Forum thread: http://tm.teresco.org/forum/index.php?topic=335 (internally only)

**Requested improvements:**
- Display errors on the site like other data checks (and replace log file)
- Integrate "nmpfps.log" to "datacheckfps.csv"
- Waypoints of the same route (and concurrent routes) should automatically be excluded
- HDX: Indicate FPs with a different color
- HDX: Option to hide FPs
- HDX: Option to hide non-FPs (show FPs only)
- HDX: Load nmp file directly (manual file download/upload must currently be done)

Feel free to add more wishes or reject my ideas. If necessary, open new issues in other repositories, e.g. data processing.

**Comment by yakra** [comment]
I opened another issue here: https://github.com/TravelMapping/EduTools/issues/69
Most of your HDX improvements are also ones I've requested; I'm glad you included them. :) (I hadn't thought of the Show FPs Only option, but yeah, it could be useful.)

**Crossposted to the forum**:
I'm hesitant to integrate NMP checks into the main datacheck. As Jim said, "they don't fit nicely into the existing structure to handle datacheck errors". With the .0005 threshold now in place, there are bound to be a huge amount in regions whose maintainers haven't been keen on removing them. This would add a LOT of noise to the main datacheck page, and make it harder to sort out the main "traditional" errors that are more of a priority to fix.
I think NMPs are better left as a lower-priority fix for those of us who want to go the extra mile and root them out. I could see bumping them up to the "datacheck" level of priority causing some burn-out and fatigue among the less active data contributors, as compared with the more gung-ho among us (such as you, me, si, or mapcat). Worst case, I could see someone taking a look at the amount of noise in datacheck and writing it off as just a bunch of busywork.
I vote for treating NMP as an optional extra, as with UnusedAltLabels.

**Comment by michihdeu** [comment]
> I opened another issue here: TravelMapping/EduTools#69

Sorry, I was quite sure that there is an issue but I couldn't find it. We have a lot of repositories and many issues could be put into Web, DataProcessing or EduTools. I've only checked Web and DataProcessing but forgot that we have EduTools too...

**Comment by yakra** [comment]
Yeah, that's a bit outside the area I pay the most attention to, but seems to be the place to go for HDX development & issues.

**Comment by jteresco** [comment]
What are the current thoughts on this, given changes in other parts of the system over the past year?

**Comment by michihdeu** [comment]
I think that this is still the **most important* feature for contributors** because I think that we need to eliminate all unintended NMPs. I only fix NMPs if I find them in the HB but don't use the current NMP files/tools because it's too cumbersome.

*Except of wpt editor which is "available" now

**Comment by yakra** [comment]
My wish list pretty much centered around flagging false-positive NMPs in the HDX and displaying them differently. This is less urgent for me now:

With my big "Operation NMPs" all completed, I don't do much of this kind of work any more, just the odd isolated case.

While I *was* doing "Operation NMPs", my work method evolved during the process: I stopped loading the [NMP files](http://forum.travelmapping.net/index.php?topic=335.msg4379#msg4379) into the HDX. Instead, I loaded them into a text editor, and looked up the points in the corresponding TMG in the HDX. (This was more useful for cases with >2 routes, such as with concurrencies.)

Moving forward, I see myself using [tmg2html](https://github.com/yakra/tmtools/tree/master/tmg2html) [pages](http://yakra.teresco.org/tmtools_demos/tmg2html/) more than the HDX itself.
If this is still a big [issue](https://github.com/yakra/tmtools/issues/69) for me (or anyone else who may start using tmg2html), I can implement detection of NMPs and false positives there.
